interfaces/acados_matlab_octave/sim_create.c

```diff
@@ -60,34 +60,32 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
 
     /* LHS */
     // field names of output struct
-    char *fieldnames[6];
+    char *fieldnames[7];
     fieldnames[0] = (char*)mxMalloc(50);
     fieldnames[1] = (char*)mxMalloc(50);
     fieldnames[2] = (char*)mxMalloc(50);
     fieldnames[3] = (char*)mxMalloc(50);
     fieldnames[4] = (char*)mxMalloc(50);
     fieldnames[5] = (char*)mxMalloc(50);
-    // TODO: add field method or plan to remember which integrator was created
-    // fieldnames[6] = (char*)mxMalloc(50);
+    fieldnames[6] = (char*)mxMalloc(50);
 
     memcpy(fieldnames[0],"config",sizeof("config"));
     memcpy(fieldnames[1],"dims",sizeof("dims"));
     memcpy(fieldnames[2],"opts",sizeof("opts"));
     memcpy(fieldnames[3],"in",sizeof("in"));
     memcpy(fieldnames[4],"out",sizeof("out"));
     memcpy(fieldnames[5],"solver",sizeof("solver"));
-
-    // memcpy(fieldnames[6],"method",sizeof("method"));
-
+    memcpy(fieldnames[6],"method",sizeof("method"));
```

Comment:
What is method? Is it the same as 'plan' for the ocp?

Reply:
For `sim` the `plan` is destroyed after this function.
So, I dont store a pointer to a C object but instead the integer corresponding to the `enum` of the `sim_solver_t`.